h2/src/main/org/h2/mvstore/MVStore.java

```diff
@@ -1475,21 +1485,26 @@ private ChunkIdsCollector getChild() {
             if (child == null) {
                 child = new ChunkIdsCollector(this);
             } else {
-                child.referenced.clear();
+                child.referencedChunks.clear();
             }
             return child;
         }
 
-        private void register(int chunkId) {
-            if (referenced.add(chunkId) && parent != null) {
-                parent.register(chunkId);
+        /**
+         * @return true if already registered
+         */
+        private boolean registerChunk(int chunkId) {
+            boolean alreadyVisited = !referencedChunks.add(chunkId); 
+            if (!alreadyVisited && parent != null) {
```

Comment:
May be use simpler logic like
```Java
if (!referencedChunks.add(chunkId)) {
    return true;
}
if (parent != null) {
```
?

Also it's strange to return `true` when operation was failed, but it does not really matter.